keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,149 @@
+
+### Non-Goals
+
+Consider any scope for filtering `manageJobsWithoutQueueName` that is not based on namespaces.
+
+## Proposal
+
+We add a `manageAllJobsNamespaceSelector` of type `*metav1.LabelSelector`
+to the top level of the Kueue `Configuration` struct (same level as `manageJobsWithoutQueueName`).
+(The name of `manageJobsWithoutQueueNameNamespaceSelector` reads poorly, so I propose something else)
+
+We use this new configuration for all integrations in the same way that the current
+`namespaceSelector` is used by the Pod integration.
+
+We migrate away from using `podOptions.namespaceSelector` over the course of several releases.
+Ideally we remove it from the `Configuration` struct as part of going to the `v1` API level.
+
+### User Stories (Optional)
+
+#### Story 1
+
+Cluster admins will be able to use `manageAllJobsNamespaceSelector` to easily configure Kueue
+to enable quota enforcement on a per namespace basis.
+
+## Design Details
+
+### API
+
+The `Configuration` struct is extended to add ManageAllJobsNamespaceSelector
+```go
+type Configuration struct {
+
+  ...
+
+	// ManageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the label kueue.x-k8s.io/queue-name.
+	// If set to true, then those jobs will be suspended and never started unless
+	// they are assigned a queue and eventually admitted. This also applies to
+	// jobs created before starting the kueue controller.
+	// Defaults to false; therefore, those jobs are not managed and if they are created
+	// unsuspended, they will start immediately.
+	ManageJobsWithoutQueueName bool `json:"manageJobsWithoutQueueName"`
+
+  // ManageAllJobsNamespaceSelector can be used to omit some namespaces from ManageJobsWithoutQueueName
+	ManageAllJobsNamespaceSelector *metav1.LabelSelector `json:"manageAllJobsNamespaceSelector,omitempty"`
+
+  ...
+}
+```
+
+### Implementation
+
+The configuration of all Job webhooks are extended to propagate the label selector and it is
+used to filter manageJobsWithoutQueueNames as is currently done in the Pod integration.
+
+Configuration parsing during startup is extended to give a deprecation notice if `podOptions.namespaceSelector` is set.
+Setting both `podOptions.namespaceSelector` and `manageAllJobsNamespaceSelector` will be flagged as a configuration error.
+
+### Test Plan
+
+[ X ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Unit Tests
+
+Unit tests for configuration parsing will be added.
+
+All unit tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+#### Integration tests
+
+All integration tests for the current Pod integration namespaceSelector will be extended across all integrations.
+
+### Graduation Criteria
+
+The feature is implemented as described for all Job integrations.
```

Comment:
I don't think this really answers the question for graduation critiera.

This would usually be what would be required to "graduate" this feature to GA.

I'm not sure if this needs a beta stage..

Reply:
I dropped this section.